subprojects/language-java/src/main/java/org/gradle/external/javadoc/StandardJavadocDocletOptions.java

```diff
@@ -76,48 +116,58 @@
-        serialWarn = addBooleanOption("serialwarn");
-        charSet = addStringOption("charset");
-        docEncoding = addStringOption("docencoding");
-        keyWords = addBooleanOption("keywords");
-        tags = addMultilineStringsOption("tag");
-        taglets = addMultilineStringsOption("taglet");
-        tagletPath = addPathOption("tagletpath");
-        docFilesSubDirs = addBooleanOption("docfilessubdirs");
-        excludeDocFilesSubDir = addStringsOption("excludedocfilessubdir", ":");
-        noQualifiers = addStringsOption("noqualifier", ":");
-        noTimestamp = addBooleanOption("notimestamp", true);
-        noComment = addBooleanOption("nocomment");
+        destinationDirectory = addFileOption(OPTION_D);
+        use = addBooleanOption(OPTION_USE);
+        version = addBooleanOption(OPTION_VERSION);
+        author = addBooleanOption(OPTION_AUTHOR);
+        splitIndex = addBooleanOption(OPTION_SPLITINDEX);
+        header = addStringOption(OPTION_HEADER);
+        windowTitle = addStringOption(OPTION_WINDOWTITLE);
+        docTitle = addStringOption(OPTION_DOCTITLE);
+        footer = addStringOption(OPTION_FOOTER);
+        bottom = addStringOption(OPTION_BOTTOM);
+        links = addMultilineStringsOption(OPTION_LINK);
+        linksOffline = addOption(new LinksOfflineJavadocOptionFileOption(OPTION_LINKOFFLINE, Lists.newArrayList()));
+        linkSource = addBooleanOption(OPTION_LINKSOURCE);
+        groups = addOption(new GroupsJavadocOptionFileOption(OPTION_GROUP, Maps.newLinkedHashMap()));
+        noDeprecated = addBooleanOption(OPTION_NODEPRECATED);
+        noDeprecatedList = addBooleanOption(OPTION_NODEPRECATEDLIST);
+        noSince = addBooleanOption(OPTION_NOSINCE);
+        noTree = addBooleanOption(OPTION_NOTREE);
+        noIndex = addBooleanOption(OPTION_NOINDEX);
+        noHelp = addBooleanOption(OPTION_NOHELP);
+        noNavBar = addBooleanOption(OPTION_NONAVBAR);
+        helpFile = addFileOption(OPTION_HELPFILE);
+        stylesheetFile = addFileOption(OPTION_STYLESHEETFILE);
+        serialWarn = addBooleanOption(OPTION_SERIALWARN);
+        charSet = addStringOption(OPTION_CHARSET);
+        docEncoding = addStringOption(OPTION_DOENCODING);
+        keyWords = addBooleanOption(OPTION_KEYWORDS);
+        tags = addMultilineStringsOption(OPTION_TAG);
+        taglets = addMultilineStringsOption(OPTION_TAGLET);
+        tagletPath = addPathOption(OPTION_TAGLETPATH);
+        docFilesSubDirs = addBooleanOption(OPTION_DOCFILESSUBDIRS);
+        excludeDocFilesSubDir = addStringsOption(OPTION_EXCLUDEDOCFILESSUBDIR, ":");
+        noQualifiers = addStringsOption(OPTION_NOQUALIFIER, ":");
+        noTimestamp = addBooleanOption(OPTION_NOTIMESTAMP, true);
+        noComment = addBooleanOption(OPTION_NOCOMMENT);
+
+        knownStandardOptionNames = Collections.unmodifiableSet(new HashSet<>(Sets.difference(optionFile.getOptions().keySet(), knownCoreOptionNames)));
```

Comment:
This difference is unnecessary, I would just inherit the core options into the set, along with a change in implementation below.
```suggestion
        knownStandardOptionNames = ImmutableSet.copyOf(optionFile.getOptions().keySet(), knownCoreOptionNames);
```

You might even consider just having a single `knownOptionNames` field that all subclasses of `CoreJavadocOptions` can use, to save memory. However, it couldn't be made `final`.

Reply:
There isn't a method like this in `ImmutableSet` in the version of Guava we're using, also, Guava collections.

Reply:
Also, aside from the concerned about it being mutable, I think it's worthwhile to keep the core options separate from the standard doclet options, in case of classes dealing with options from non-standard doclets.